### PR TITLE
MGMT-4468 update cluster from installenv

### PR DESCRIPTION
**Pull request by filanov**
Update cluster from InstallEnv when setting Proxy and NTP conficution
Those parameters are reletaed only to discovery phase so they should not affect the cluster
Instead of having a ping-pong between cluster deployment controller and install env controller it is done once via InstallEnv controller.
Later when image will be separate from the cluster it will make things simple.

### CODE REVIEW

File: internal/controller/controllers/clusterdeployments_controller.go [hunk 1]
```diff
@@ -296,17 +286,6 @@ func (r *ClusterDeploymentsReconciler) createNewCluster(
 		UserManagedNetworking: swag.Bool(isUserManagedNetwork(clusterDeployment)),
 	}
 
-	installEnv, err := getInstallEnvByClusterDeployment(ctx, r.Client, clusterDeployment)
```
Comment (nmagnezi) [inline internal/controller/controllers/clusterdeployments_controller.go:299]: How will newly created `clusterDeployments` get those params? in the next iteration of InstallEnv reconcile?

Comment (filanov) [reply internal/controller/controllers/clusterdeployments_controller.go:299]: You need it only when creating an ISO so new cluster don't really have a use for those fields.

**Review by nmagnezi — APPROVED** [review]

**Review by nmagnezi — COMMENTED** [review]
/test e2e-metal-assisted

**Pull request merged**